privacyIDEA Authenticator/Crypto.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  Crypto.swift
+//  privacyIDEA Authenticator
+//
+//  Created by Nils Behlen on 10.05.19.
+//  Copyright © 2019 Nils Behlen. All rights reserved.
+//
+
+import Foundation
+import SwiftyRSA
+import SwiftOTP
+
+class Crypto {
+    
+    static let shared = Crypto()
+    
+    private init () {}
+    
+    // Generate a Keypair, which is stored with the serial as alias
+    // Return the PublicKey as b64String
+    func generateKeypair(_ serial: String) -> String {
+        U.log("Generating Keypair for serial \(serial)...")
+        var error: Unmanaged<CFError>?
+        guard let privateKey = SecKeyCreateRandomKey(Constants.keyPairAttr as CFDictionary, &error) else {
+            U.log("Error while generating keys:")
+            U.log(error!.takeRetainedValue() as Error)
+            return ""
```

Comment:
What is the best way to do error handling?
Is there a problem in returning an empty string or further below "nil", if something goes wrong?
Could this be a possible attack vector, modifying the string representation of the key and thus create a nil response?

Is the empty string or nil handled in the upper level?

Reply:
the empty string would be sent to pi. 
It is probably better if the caller of this (push-rollout) fails if there is no key generated.

Reply:
I think we can go with this at the moment.
privacyIDEA will complain, that the enrollment failed.
But after all, the key generation failed, so the error could be a bit misleading.
Can you please open an issue, so that we remember to someday^(TM) fix this?

Reply:
I changed the method to return nil on failure, so the push rollout gets cancelled early.
But i also opened an issue to check all the error cases again